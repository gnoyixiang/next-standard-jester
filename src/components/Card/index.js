import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const StyledCard = styled.a`
  margin: 1rem;
  flex-basis: 45%;
  padding: 1.5rem;
  text-align: left;
  color: inherit;
  text-decoration: none;
  border: 1px solid #eaeaea;
  border-radius: 10px;
  transition: color 0.15s ease, border-color 0.15s ease;

  :hover,
  :focus,
  :active {
    color: #0070f3;
    border-color: #0070f3;
  }

  h3 {
    margin: 0 0 1rem 0;
    font-size: 1.5rem;
  }

  p {
    margin: 0;
    font-size: 1.25rem;
    line-height: 1.5;
  }
`

const Card = props => {
  const { title, description, link, ...restProps } = props
  return (
    <StyledCard href={link} {...restProps}>
      <h3>{title} &rarr;</h3>
      <p>{description}</p>
    </StyledCard>
  )
}

Card.propTypes = {
  title: PropTypes.any,
  link: PropTypes.string,
  description: PropTypes.any
}

export default Card
