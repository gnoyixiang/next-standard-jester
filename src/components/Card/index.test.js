/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'
import Card from '.'

describe('Card', () => {
  test('renders inactive without crashing', () => {
    const wrapper = shallow(<Card />)
    expect(wrapper).toMatchSnapshot()
  })
})
