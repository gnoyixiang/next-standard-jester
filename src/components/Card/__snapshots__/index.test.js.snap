// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders inactive without crashing 1`] = `
<Card__StyledCard>
  <h3>
     â†’
  </h3>
  <p />
</Card__StyledCard>
`;
